dataclients/kubernetes/kube_test.go

```diff
@@ -2931,7 +3341,258 @@ func TestSkipperCustomRoutes(t *testing.T) {
 			defer api.Close()
-			dc, err := New(Options{KubernetesURL: api.server.URL})
+			dc, err := New(Options{
+				KubernetesURL: api.server.URL,
+			})
+			if err != nil {
+				t.Error(err)
+			}
+
+			defer dc.Close()
+
+			r, err := dc.LoadAll()
+			if err != nil {
+				t.Error(err)
+				return
+			}
+
+			checkPrettyRoutes(t, r, ti.expectedRoutes)
+		})
+	}
+}
+
+func TestSkipperCustomRoutesEastWest(t *testing.T) {
+	for _, ti := range []struct {
+		msg            string
+		services       services
+		ingresses      []*ingressItem
+		expectedRoutes map[string]string
+	}{{
+		msg: "ingress with 1 host definitions and 1 additional custom route",
+		services: services{
+			"foo": map[string]*service{
+				"bar": testService("1.2.3.4", map[string]int{"baz": 8181}),
+			},
+		},
+		ingresses: []*ingressItem{testIngress("foo", "qux", "", "", "", "",
+			`Method("OPTIONS") -> <shunt>`,
+			"", backendPort{}, 1.0,
+			testRule("www1.example.org", testPathRule("/", "bar", backendPort{"baz"})),
+		)},
+		expectedRoutes: map[string]string{
+			"kube_foo__qux__www1_example_org_____bar":   "Host(/^www1[.]example[.]org$/) && PathRegexp(/^\\//) -> \"http://1.2.3.4:8181\"",
+			"kube_foo__qux__0__www1_example_org_____":   "Host(/^www1[.]example[.]org$/) && PathRegexp(/^\\//) && Method(\"OPTIONS\") -> <shunt>",
+			"kubeew_foo__qux__www1_example_org_____bar": "Host(/^qux[.]foo[.]skipper[.]cluster[.]local$/) && PathRegexp(/^\\//) -> \"http://1.2.3.4:8181\"",
+			"kubeew_foo__qux__0__www1_example_org_____": "Host(/^qux[.]foo[.]skipper[.]cluster[.]local$/) && PathRegexp(/^\\//) && Method(\"OPTIONS\") -> <shunt>",
+		},
+	}, {
+		msg: "ingress with 1 host definitions with path and 1 additional custom route",
+		services: services{
+			"foo": map[string]*service{
+				"bar": testService("1.2.3.4", map[string]int{"baz": 8181}),
+			},
+		},
+		ingresses: []*ingressItem{testIngress("foo", "qux", "", "", "", "",
+			`Method("OPTIONS") -> <shunt>`,
+			"", backendPort{}, 1.0,
+			testRule("www1.example.org", testPathRule("/a/path", "bar", backendPort{"baz"})),
+		)},
+		expectedRoutes: map[string]string{
+			"kube_foo__qux__www1_example_org___a_path__bar":   "Host(/^www1[.]example[.]org$/) && PathRegexp(/^\\/a\\/path/) -> \"http://1.2.3.4:8181\"",
+			"kube_foo__qux__0__www1_example_org_a_path____":   "Host(/^www1[.]example[.]org$/) && PathRegexp(/^\\/a\\/path/) && Method(\"OPTIONS\") -> <shunt>",
+			"kube___catchall__www1_example_org____":           "Host(/^www1[.]example[.]org$/) -> <shunt>",
+			"kubeew_foo__qux__www1_example_org___a_path__bar": "Host(/^qux[.]foo[.]skipper[.]cluster[.]local$/) && PathRegexp(/^\\/a\\/path/) -> \"http://1.2.3.4:8181\"",
+			"kubeew_foo__qux__0__www1_example_org_a_path____": "Host(/^qux[.]foo[.]skipper[.]cluster[.]local$/) && PathRegexp(/^\\/a\\/path/) && Method(\"OPTIONS\") -> <shunt>",
+			//"kubeew___catchall__www1_example_org____":         "Host(/^qux[.]foo[.]skipper[.]cluster[.]local$/) -> <shunt>",
```

Comment:
i think we need the catchall for the east-west routes, as well. Is it WIP?

Reply:
yes

Reply:
I added a todo list